File: eticket-backend/src/main/java/com/mkostadinov/eticketbackend/repository/VehicleRepository.java
package com.mkostadinov.eticketbackend.repository;

import com.mkostadinov.eticketbackend.model.entity.Vehicle;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface VehicleRepository extends JpaRepository<Vehicle, String> {

    boolean existsByRegistrationNumber(String registrationNumber);

    Optional<Vehicle> findByRegistrationNumber(String registrationNumber);
}

File: eticket-kat-api/src/main/kotlin/com/eticketkatsimulator/repository/VehicleRepository.kt
package com.eticketkatsimulator.repository

import com.eticketkatsimulator.model.entity.Vehicle
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import org.springframework.stereotype.Repository
import reactor.core.publisher.Flux

@Repository
interface VehicleRepository : ReactiveMongoRepository<Vehicle, String> {
    fun findAllByOwnerDriverLicenseId(ownerId: String): Flux<Vehicle>;

}